components/camel-jms/src/main/java/org/apache/camel/component/jms/JmsConfiguration.java

```diff
@@ -620,7 +620,13 @@ private Object doSendToDestination(
                 // Check commit - avoid commit call within a JTA transaction.
                 if (session.getTransacted() && isSessionLocallyTransacted(session)) {
                     // Transacted session created by this template -> commit.
-                    JmsUtils.commitIfNecessary(session);
+                    Assert.notNull(session, "Session must not be null");
+                    try {
+                        session.commit();
```

Comment:
Why are you not calling JmsUtils.commitIfNessasary ?

Reply:
@davsclaus take a look at jira description, it's all there.
https://issues.apache.org/jira/browse/CAMEL-20521
I'll probably open a PR on spring as well.

Reply:
No camel-jms is for JMS and should be as-is that is battle tested with spring-jms and all those kind of JMS brokers.

This ticket is about AMQP which is not JMS.
If there is a special need for camel-amqp in AMQP mode, then that must be done specially in camel-amqp and not in general camel-jms.

Spring likely dont care about your AMQP as they use rabbitmq and its AMQP version.

Reply:
IMHO, a more appropriate fix for this would involve creating an specific JMS template for AMQP.

At a quick glace, it doesn't seem like the code easily allows this, so probably would involve some refactoring on the process.

Reply:
Ok, I'll try to reproduce the problem with plain jms component, if it's only a AMQP/QPid problem, I'll create a JmsTemplate for AMQP.

Reply:
@davsclaus @orpiske, not able to reproduce with plain jms, it seems that this happens only with amqp/Qpid. I moved the changes to camel-amqp component.